lib/hostip.c

```diff
@@ -652,13 +652,31 @@ enum resolve_t Curl_resolv(struct Curl_easy *data,
   CURLcode result;
   enum resolve_t rc = CURLRESOLV_ERROR; /* default to failure */
   struct connectdata *conn = data->conn;
+  /* We should intentionally error and not resolve .onion TLDs */
+  bool found_onion = FALSE;
+  char *last_dot = strrchr(hostname, '.');
+  if(last_dot) {
+    char *stripped_hostname = strdup(hostname);
+    if(strlen(last_dot) == 1) {
+      /* We probably got the trailing dot on a FQDN */
+      stripped_hostname[strlen(hostname)-1] = '\0';
+      last_dot = strrchr(stripped_hostname, '.');
+    }
+    if(last_dot && curl_strequal(".onion", last_dot)) {
```

Comment:
Are we at this point certain we don't have multibyte chars in `last_dot` such that `curl_strequal` works as intended? It might not be a problem in practice, but comparing apples to oranges have bitten code in the past.